### PR TITLE
Added an API call to the db and local file storage

### PR DESCRIPTION
**Pull request by aidan-le-beard**
I know we might not want the Dart API call, but I'm thinking the local storage is something we definitely do want, and we can take out the API call / dependency if needed. So I'm thinking this should be main, before you start testing on iOS and implementing Flask :D

### CODE REVIEW

File: sp5GreenFlutterDartAPIAndLocalFileSaving/lib/main.dart [hunk 1]
```diff
@@ -1,20 +1,101 @@
...
+  // get the phone's path to where it saves files for this app
+  Directory appDocumentsDirectory = await getApplicationDocumentsDirectory();
+  var appDocumentsPath = appDocumentsDirectory.path;
+  filePath = "$appDocumentsPath";
+
+  // loggedIn boolean NOT IMPLEMENTED YET
+  if (loggedIn) {
+    /** get groceryList from API, check what group they're in...
+        ACTUALLY, don't we want each user to have their own local list?
+        SO, we probably don't want this.
```
Comment (quindariuss) [inline sp5GreenFlutterDartAPIAndLocalFileSaving/lib/main.dart:44]: Correct!

File: sp5GreenFlutterDartAPIAndLocalFileSaving/lib/main.dart [hunk 2]
```diff
@@ -1,20 +1,101 @@
...
+        API calls should actually only happen when... clicking Add to Group List / clicking Generate Shopping List **/
+  }
+
+  // this willl likely be changed to the first IF, and above removed
+  // else check if there's a local file, and use that
+  else if (File(filePath + "/groceryItems.txt").existsSync() && File(filePath + "/groceryItemsCopy.txt").existsSync()) {
+    // convert the .txt json and save in _groceryItems
```
Comment (quindariuss) [inline sp5GreenFlutterDartAPIAndLocalFileSaving/lib/main.dart]: Save the document as JSON you animal 🦒

File: sp5GreenFlutterDartAPIAndLocalFileSaving/lib/main.dart (hunk 1 above)
Comment (aidan-le-beard) [reply sp5GreenFlutterDartAPIAndLocalFileSaving/lib/main.dart:44]: We'll probably still need a loggedIn bool for the shared part of the list later on, once we get to where we can implement it

File: sp5GreenFlutterDartAPIAndLocalFileSaving/lib/main.dart (hunk 2 above)
Comment (aidan-le-beard) [reply sp5GreenFlutterDartAPIAndLocalFileSaving/lib/main.dart]: 😂 Sir yes sir, back over to you, sir.

**Review by quindariuss — COMMENTED** [review]
Nice Icon!

**Comment by quindariuss** [comment]
I think this branch is ready to go!
As a general rule the person that codes it out merges it, and you have my approval for sure!

**Review by quindariuss — APPROVED** [review]
Ready to go!

**Pull request merged**